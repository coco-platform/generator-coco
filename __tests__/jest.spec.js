/**
 * @description - coco:jest generator
 * @author - huang.jian <hjj491229492@hotmail.com>
 */

// packages
const path = require('path');
const fs = require('fs');
const yeomon = require('yeoman-test');
const assert = require('yeoman-assert');
const hbs = require('handlebars');

describe('coco:jest generator', () => {
  const template = fs.readFileSync(
    path.resolve(
      process.cwd(),
      'generators',
      'jest',
      'templates',
      'jest.config.js.hbs'
    ),
    'utf8'
  );
  const compile = hbs.compile(template);

  it('should properly render template with universal require', () => {
    const context = {
      universal: true,
      root: 'native',
      environment: 'node',
    };

    expect(compile(context)).toMatchSnapshot();
  });

  it('should properly render template without universal require', () => {
    const context = {
      universal: false,
      root: 'native',
      environment: 'node',
    };

    expect(compile(context)).toMatchSnapshot();
  });

  it('should output jest configuration', () => {
    const jest = path.resolve(__dirname, '../generators/jest');
    const prompts = {
      universal: false,
      environment: 'node',
      root: './src',
    };

    return yeomon
      .run(jest)
      .withPrompts(prompts)
      .then(() => {
        assert.file(['jest.config.js']);

        const configuration = fs.readFileSync('jest.config.js', 'utf8');

        expect(configuration).toMatchSnapshot();
      });
  });
});
