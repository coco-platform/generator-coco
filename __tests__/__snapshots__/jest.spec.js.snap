// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coco:jest generator should output jest configuration 1`] = `
"module.exports = {
  transform: {
    '\\\\\\\\.(ts|tsx|js|jsx)$': 'babel-jest',
  },
  moduleFileExtensions: ['js', 'jsx', 'json', 'ts', 'tsx'],
  moduleDirectories: ['node_modules'],
  collectCoverageFrom: ['./src/**/*.{js,jsx,ts,tsx}'],
  coverageDirectory: 'coverage',
  coverageReporters: ['lcov', 'html'],
  transformIgnorePatterns: ['/node_modules'],
  testEnvironment: 'node',
  testMatch: ['<rootDir>/__tests__/**/*.spec.{js,jsx,ts,tsx}'],
  testPathIgnorePatterns: ['/node_modules/'],
};
"
`;

exports[`coco:jest generator should properly render template with universal require 1`] = `
"module.exports = {
  transform: {
    '\\\\\\\\.(ts|tsx|js|jsx)$': 'babel-jest',
    '\\\\\\\\.(jpg|jpeg|png|giff|webp|svg)$':
      '@coco-platform/jest-tools/lib/asset-transformer.js',
  },
  moduleFileExtensions: ['js', 'jsx', 'json', 'ts', 'tsx'],
  moduleDirectories: ['node_modules'],
  moduleNameMapper: {
    '\\\\\\\\.css$': '@coco-platform/jest-tools/lib/css-ignore.js',
    '\\\\\\\\.pcss': '@coco-platform/jest-tools/lib/css-modules.js',
  },
  collectCoverageFrom: ['native/**/*.{js,jsx,ts,tsx}'],
  coverageDirectory: 'coverage',
  coverageReporters: ['lcov', 'html'],
  transformIgnorePatterns: ['/node_modules'],
  testEnvironment: 'node',
  testMatch: ['<rootDir>/__tests__/**/*.spec.{js,jsx,ts,tsx}'],
  testPathIgnorePatterns: ['/node_modules/'],
};
"
`;

exports[`coco:jest generator should properly render template without universal require 1`] = `
"module.exports = {
  transform: {
    '\\\\\\\\.(ts|tsx|js|jsx)$': 'babel-jest',
  },
  moduleFileExtensions: ['js', 'jsx', 'json', 'ts', 'tsx'],
  moduleDirectories: ['node_modules'],
  collectCoverageFrom: ['native/**/*.{js,jsx,ts,tsx}'],
  coverageDirectory: 'coverage',
  coverageReporters: ['lcov', 'html'],
  transformIgnorePatterns: ['/node_modules'],
  testEnvironment: 'node',
  testMatch: ['<rootDir>/__tests__/**/*.spec.{js,jsx,ts,tsx}'],
  testPathIgnorePatterns: ['/node_modules/'],
};
"
`;
